app/src/org/commcare/dalvik/odk/provider/FormsProvider.java

```diff
@@ -106,9 +118,12 @@ public boolean onCreate() {
     }
     
     public void init() {
-        //this is terrible, we need to be binding to the cc service, etc. Temporary code for testing
-        if(mDbHelper == null) {
-            mDbHelper = new DatabaseHelper(CommCareApplication._(), DATABASE_NAME);
+        // this is terrible, we need to be binding to the cc service, etc. Temporary code for testing
+        if (mDbHelper == null || mDbHelper.getAppId() != ProviderUtils.getSeatedOrInstallingAppId()) {
```

Comment:
It worries me a bit that we have the seated or installed notion, because it concerns me that we may end up thinking we have both a seated and an "installing" app at the same time. 

Is there some sort of consistency checking to ensure that we don't have a seated app when we set the installing app, and that we don't have an installing app when we seat?